Dependency issue: ibm-db==3.2.7 causes exception

**Describe the bug**
DB validations fail with recently updated `ibm-db==3.2.7`:

```
FAILED tests/system/data_sources/test_db2.py::test_column_validation_core_types_to_bigquery - TypeError: Arguments differences_target_value:float64 and differences_source_value:string are not comparable
```

They were originally failing with the error below but that is resolved by changes on the attached branch: 
```
E               ibm_db_dbi.ProgrammingError: ibm_db_dbi::ProgrammingError: Statement Execute Failed: [IBM][CLI Driver][DB2/LINUXX8664] SQL0206N  "TIMEZONE" is not valid in the context where it is used.  SQLSTATE=42703 SQLCODE=-206
```

https://pypi.org/project/ibm-db/3.2.7/

When this issue is complete we can remove the pinned version from noxfile.py.


**What version of DVT are you using?**
8.2.0(dev)

**What type of connections are you using for source and target?**
DB2

**Steps to reproduce the behavior**
```
pip install --upgrade ibm-db
```

I retested this and now get a different exception:

```
>           raise TypeError(
                f'Arguments {rlz._arg_type_error_format(left)} and '
                f'{rlz._arg_type_error_format(right)} are not comparable'
            )
E           TypeError: Arguments differences_target_value:float64 and differences_source_value:string are not comparable
```

Tested with 3.2.8 and tests pass.

--- a/noxfile.py
+++ b/noxfile.py
@@ -309,9 +309,9 @@ def integration_db2(session):
     """Run DB2 integration tests.
     Ensure DB2 validation is running as expected.
     """
-    # TODO Remove dependency "ibm-db<3.2.7" below when working on issue-1591.
+    # Dependency "ibm-db>=3.2.8" because 3.2.7 causes an exception in DVT (issue-1591).
     _setup_session_requirements(
-        session, extra_packages=["ibm-db-sa<0.4.2", "ibm-db<3.2.7"]
+        session, extra_packages=["ibm-db-sa<0.4.2", "ibm-db>=3.2.8"]
     )
 
     expected_env_vars = [

--- a/noxfile.py
+++ b/noxfile.py
@@ -310,7 +310,7 @@ def integration_db2(session):
     Ensure DB2 validation is running as expected.
     """
     _setup_session_requirements(session, extra_packages=["ibm-db-sa", "ibm-db>=3.2.8"])
-    
+
     expected_env_vars = [
         "PROJECT_ID",
         "DB2_HOST",

